crates/uv-distribution-types/src/index.rs

```diff
@@ -82,6 +82,44 @@ pub struct Index {
     /// publish-url = "https://upload.pypi.org/legacy/"
     /// ```
     pub publish_url: Option<Url>,
+    /// The authentication policy for the index.
+    ///
+    /// All requests made to URLs with this index URL as a prefix will follow
+    /// the policy corresponding to this authentication policy.
+    ///
+    /// ```toml
+    /// [[tool.uv.index]]
+    /// name = "my-index"
+    /// url = "https://<omitted>/simple"
+    /// auth_policy = "always"
+    /// ```
+    #[serde(default)]
+    pub auth_policy: AuthPolicy,
+}
+
+#[derive(
+    Copy, Clone, Debug, Default, Hash, Eq, PartialEq, serde::Serialize, serde::Deserialize,
+)]
+#[serde(rename_all = "kebab-case")]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum AuthPolicy {
+    /// Authenticate only when necessary.
+    #[default]
+    Auto,
+    /// Always authenticate.
+    Always,
+    /// Never authenticate.
+    Never,
+}
+
+impl From<AuthPolicy> for uv_auth::AuthPolicy {
+    fn from(item: AuthPolicy) -> Self {
+        match item {
+            AuthPolicy::Always => uv_auth::AuthPolicy::Always,
+            AuthPolicy::Auto => uv_auth::AuthPolicy::Auto,
+            AuthPolicy::Never => uv_auth::AuthPolicy::Never,
+        }
+    }
 }
```

Comment:
We may want to avoid repeating all of this if they're exactly equivalent. I've been hesitant to add `schemars` etc. as a dependency to the `uv-auth` crate, but I think we do that all over the place anyway.

If we _do_ retain these duplicates, the internal `uv_auth` type should refer to this type in the rustdoc so we don't need to maintain documentation in two places.

Also seems like this type might go in `uv-configuration::authentication` if we were going to expose a separate configuration-specific type.

Reply:
I had originally tried that but following the pattern I saw elsewhere, I ended up getting a `the trait `JsonSchema` is not implemented for AuthPolicy` compiler error despite deriving it. However, I ended up abandoning it when investigating further because I thought we might want to keep these `JsonSchema` structs in `uv-configuration`.

However, if you'd prefer, we can look into why that wasn't compiling so we can update.

Reply:
Oh, weird. Given the types are identical, it sounds like it's worth understanding why the derive wasn't working. I don't feel particularly strongly otherwise.

Reply:
Yeah that was it. Updated